scripts/toolchain/macos/build.sh

```
@@ -0,0 +1,138 @@
+# Usage: ./scripts/toolchain/macos/build.sh -r <ref> -p <installation directory>
+
+set -euo pipefail
+
+# Parse arguments
+while getopts "r:p:" opt; do
+  case $opt in
+    r) REF="$OPTARG"
+    ;;
+    p) INSTALL_PREFIX="$OPTARG"
+    ;;
+  esac
```

Shellcheck (0.11.0)

[warning] Invalid flags are not handled. Add a *) case.

(SC2220)